endpoints/openrtb2/amp_auction.go

```diff
@@ -230,6 +231,22 @@ func (deps *endpointDeps) AmpAuction(w http.ResponseWriter, r *http.Request, _ h
 		return
 	}
 
+	// Populate any "missing" OpenRTB fields with info from other sources, (e.g. HTTP request headers).
+	deps.setFieldsImplicitly(r, reqWrapper, account)
+
+	hasStoredResponses := len(storedAuctionResponses) > 0
+	errs := deps.validateRequest(reqWrapper, true, hasStoredResponses, storedBidResponses, false)
+	errL = append(errL, errs...)
+	ao.Errors = append(ao.Errors, errs...)
+	if errortypes.ContainsFatalError(errs) {
+		w.WriteHeader(http.StatusBadRequest)
+		for _, err := range errortypes.FatalOnly(errs) {
+			w.Write([]byte(fmt.Sprintf("Invalid request: %s\n", err.Error())))
+		}
+		labels.RequestStatus = metrics.RequestStatusBadInput
+		return
```

Comment:
Could you cover this section with a test?

Reply:
fixed `TestAmpBadRequests` and added `tmax-negative.json` for this scenario